packages/ui/src/Shared/DndFileChooser.tsx

```diff
@@ -0,0 +1,312 @@
+import {
+  Grid,
+  Icon,
+  TimedToastNotification,
+  ToastNotificationList,
+} from 'patternfly-react';
+import * as React from 'react';
+import Dropzone from 'react-dropzone';
+import { Container } from '../Layout/Container';
+import './DndFileChooser.css';
+
+/**
+ * The properties of the `DndFileChooser`.
+ */
+export interface IDndFileChooserProps {
+  /**
+   * Indicates if multiple files can be added. Defaults to `false`.
+   */
+  allowMultiple?: boolean;
+
+  /**
+   * `true` if the dropzone should be disabled. Defaults to `false`.
+   */
+  disableDropzone: boolean;
+
+  /**
+   * A comma delimited string of file extensions. For example, '.jar,.txt'. Defaults to any file extension.
+   */
+  fileExtensions?: string;
+
+  /**
+   * The localized text that appears below the selected file information at the bottom of the drop area.
+   */
+  i18nHelpMessage?: string;
+
+  /**
+   * The localized text (may include HTML tags) that appears above the selected file information at the
+   * top of the drop area.
+   */
+  i18nInstructions: string;
+
+  /**
+   * The localized text that appears when no file has been selected.
+   */
+  i18nNoFileSelectedMessage: string;
+
+  /**
+   * The localized text for the label that appears before the selected file information.
+   */
+  i18nSelectedFileLabel: string;
+
+  /**
+   * A general, localized message for when a file upload failed. This message will be shown
+   * along with an error icon and appears next to the selected file message. If this property
+   * is set then `i18nUploadSuccessMessage` should not be set.
+   */
+  i18nUploadFailedMessage?: string;
+
+  /**
+   * A general, localized message for when a file upload was successful. This message will be shown
+   * along with an OK icon and appear next to the selected file message. If this property
+   * is set then `i18nUploadFailedMessage` should not be set.
+   */
+  i18nUploadSuccessMessage?: string;
```

Comment:
I'm not sure about this; the comment says that `i18nUploadFailedMessage` and `i18nUploadSuccessMessage` are mutually exclusive, but `getUploadMessage` use both of these props depending on the state of the upload.

Shouldn't these two props be both required?

Reply:
The upload message uses the existence of those props to decide which icon to display next to the message. An alternative way would be to make them both required and add another property that indicates which message and icon to display. Is that what you had in mind?

Reply:
Ah, I see, but now I'm more confused. Shouldn't showing one message or the other depend on the state of the upload?

Reply:
Yes, it does depend on the state of the upload. The DND component accepts and rejects files based on if the file has the correct file extension and if multiple files are allowed. Once the DND component accepts or rejects the file the parent gets notified via an accept file or reject file callback. The parent can process the file(s) involved in those callbacks however they want. For instance, for extension import, when a file is accepted by the DND component, that file is sent to the backend. Based on that processing the parent can determine an upload state it wants displayed in the DND component. Does that make sense? How would you like to see this work?

Reply:
Ah, I think I get it now. It kinda feels like this logic should be handled completely by the parent component, which could simply use some other components meant to handle successful/failed uploads.

I mean, it's just an `if` from the parent component standpoint, but I think it simplifies understanding how to use this component.